numba/tests/test_np_functions.py

```diff
@@ -3181,6 +3193,109 @@ def test_split_basic(self):
         self.assertIn("np.split: Argument axis out of bounds",
                       str(raises.exception))
 
+    def test_vsplit_basic(self):
+        # split and array_split have more comprehensive tests of splitting.
+        # only do simple test on vsplit
+        def inputs():
+            # Taken from https://github.com/numpy/numpy/blob/...
+            # 11a2ea96a7ac923c2283d487bffa7fce8924cbd7/numpy/lib/tests/...
+            # test_shape_base.py#L586-L5917
+
+            # test_2D_array
+            yield np.array([[1, 2, 3, 4], [1, 2, 3, 4]]), 2
+            yield np.array([[1., 2., 3., 4.], [1., 2., 3., 4.]]), 2
+            yield np.arange(16.0).reshape(4, 4), 2
+            yield np.arange(16.0).reshape(4, 4), np.array([3, 6])
+            yield np.arange(16.0).reshape(4, 4), [3, 6]
+            yield np.arange(16.0).reshape(4, 4), (3, 6)
+            yield np.arange(8.0).reshape(2, 2, 2), 2
+
+        pyfunc = vsplit
+        cfunc = jit(nopython=True)(pyfunc)
+
+        for a, ind_or_sec in inputs():
+            self.assertPreciseEqual(pyfunc(a, ind_or_sec),
+                                    cfunc(a, ind_or_sec))
+
+    def test_hsplit_basic(self):
+        # split and array_split have more comprehensive tests of splitting.
+        # only do simple test on hsplit
+        def inputs():
+            # Taken from https://github.com/numpy/numpy/blob/...
+            # 11a2ea96a7ac923c2283d487bffa7fce8924cbd7/numpy/lib/tests/...
+            # test_shape_base.py#L553-L564
+
+            # test_1D_array
+            yield np.array([1, 2, 3, 4]), 2
+            yield np.array([1., 2., 3., 4.]), 2
+            # test_2D_array
+            yield np.array([[1, 2, 3, 4], [1, 2, 3, 4]]), 2
+            yield np.array([[1., 2., 3., 4.], [1., 2., 3., 4.]]), 2
+            yield np.arange(16.0).reshape(4, 4), 2
+            yield np.arange(16.0).reshape(4, 4), np.array([3, 6])
+            yield np.arange(16.0).reshape(4, 4), [3, 6]
+            yield np.arange(16.0).reshape(4, 4), (3, 6)
+            yield np.arange(8.0).reshape(2, 2, 2), 2
```

Comment:
Tests from hsplit/vsplit/dsplit are the same. Merge the functions into a single one